source/common/network/connection_impl.h

```diff
@@ -179,6 +217,10 @@ class ConnectionImpl : public ConnectionImplBase, public TransportSocketCallback
   bool write_end_stream_ : 1;
   bool current_write_end_stream_ : 1;
   bool dispatch_buffered_data_ : 1;
+  /**
+   * Instance to perform run-time configurable operations on the socket
```

Comment:
I think QUIC has handles for setting send/recv buffer size.
Less sure of watermark.
@danzh2010 for commentary

Reply:
QUIC doesn't have handle to set these socket options. QUIC connection buffer is throttled by connection flow control window which is not changeable during the connection. I'm not sure how to map this feature to a QUIC connection. But I think at least we can opt-out this feature by simply implement Connection::getDynamicSocketOptionsPtr() to return nullptr for a QUIC connection and HCM can decide what to do in that case. @alyssawilk, @mattklein123 Does this sounds okay? 

We probably should set the UDP listening socket buffer size at start up. This can be applied to all UDP traffic.

Reply:
Per my above comment it's unclear why we need to return a pointer here at all vs. just having these be interface functions that are implemented or not?